pkg/transform/kubernetes/populators.go

```diff
@@ -161,3 +161,34 @@ func populateContainers(containers []spec.Container, cms []spec.ConfigMapMod, se
 	log.Debugf("containers after populating health: %s", string(b))
 	return cnts, nil
 }
+
+// Since we are automatically creating pvc from
+// root level persistent volume and entry in the container
+// volume mount, we also need to update the pod's volume field
+func populateVolumes(containers []api_v1.Container, volumeClaims []spec.VolumeClaim,
```

Comment:
you don't have to pass pointer, and you can still return new `[]api_v1.Volume` but passing whole `spec.App` is going to be much better than providing 3 arguments that are all in the same base struct. It will be also much less error prone for anyone calling this function